pyprophet/levels_contexts.py

```diff
@@ -184,7 +189,7 @@ def infer_proteins(infile, outfile, context, parametric, pfdr, pi0_lambda, pi0_m
     con.close()
 
     if context == 'run-specific':
-        data = data.groupby('run_id').apply(statistics_report, outfile, context, "protein", parametric, pfdr, pi0_lambda, pi0_method, pi0_smooth_df, pi0_smooth_log_pi0, lfdr_truncate, lfdr_monotone, lfdr_transformation, lfdr_adj, lfdr_eps, color_palette).reset_index()
+        data = data.groupby('run_id').apply(statistics_report, outfile, context, "protein", parametric, pfdr, pi0_lambda, pi0_method, pi0_smooth_df, pi0_smooth_log_pi0, lfdr_truncate, lfdr_monotone, lfdr_transformation, lfdr_adj, lfdr_eps, color_palette)
```

Comment:
Removing reset index is required to prevent the error 
```
  File "/home/joshua/mambaforge/envs/pyprophet_dev/lib/python3.11/site-packages/pandas/core/frame.py", line 5158, in insert
    raise ValueError(f"cannot insert {column}, already exists")
ValueError: cannot insert run_id, already exists
```

Same as below. Must be a change to pandas groupby functionality

Reply:
Yeah seems like it some groupby deprecations occured for [Pandas v2.2.0](https://pandas.pydata.org/docs/whatsnew/v2.2.0.html)

> Deprecated the Grouping attributes group_index, result_index, and group_arraylike; these will be removed in a future version of pandas ([GH 56148](https://github.com/pandas-dev/pandas/issues/56148))

If you don't mind, would you be able to test with a version prior to pandas v2.2.0, to see if the old code works with the `.reset_index()`, just so we know for sure that is the change.